src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs

```diff
@@ -8,6 +8,32 @@
 
 namespace Microsoft.ML
 {
+    /// <summary>
+    /// Specifies input and output column names for a transformation.
+    /// </summary>
+    public sealed class InputOutputColumnPair
```

Comment:
>InputOutputColumnPair [](start = 24, length = 21)

what is difference between this one and `ColumnOptions`?

Reply:
Not much, I would have used the bellow if it were me, but Tom asked it to be different, for some reason!

---
In reply to: [268334077](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268334077) [](ancestors = 268334077)

Reply:
Yes, a very good reason. `ColumnOptions` is a struct meant to serve a *specific* transformer base, and that is involved in the type heirarchy, and in particular something that captured all of the individual settings and state for each mapping. Our goals here were comparatively more modest: we just needed to . Well designed code does what it is designed to do, and in the simplest possible way. There is no need for this to be part of an elaborate type hierarchy -- this was in fact the mistake that led to the issue #2884 being filed, that we'd conflated two *distinct* techniques for the extremely bad reasoning that they both had to do with "column." (And, of course, if something deals with the same *sort* of object, obviously they belong in the same type heirarchy, right?)

---
In reply to: [268334389](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268334389) [](ancestors = 268334389,268334077)

Reply:
Just in case, I'm not talking about `ColumnOptions` inside Estimator, (for example `HashingEstimator.ColumnOptions`.
I'm talking about `ColumnOptions` in this exact file few lines below (line 41)
Only difference I see is Input/output column names is public in this one instead of private in other, and name of class.
Ok, two difference, one below has implicit converter from tuple.

Can we delete `ColumnOptions` in this file?

---
In reply to: [268337035](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268337035) [](ancestors = 268337035,268334389,268334077)

Reply:
*That's* fine. What Artidoro and I had discussed was actually somewhat different. (Or we were talking about two separate things without realizing it.)

---
In reply to: [268339442](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268339442) [](ancestors = 268339442,268337035,268334389,268334077)

Reply:
I guess we were talking about something different, glad we are on the same page.

---
In reply to: [268760195](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268760195) [](ancestors = 268760195,268339442,268337035,268334389,268334077)

Reply:
This seems good to me. The `ColumnOptions` I had been talking about (which is to say, the vast majority of things with that name) are as I said meant to serve a different purpose.

We should probably rename them back to `ColumnInfo` at some point but this can be delayed as they are internal... ummm except one. Whoops. Opened #3078. :) Aside from that, yeah.

---
In reply to: [268771510](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268771510) [](ancestors = 268771510,268760195,268339442,268337035,268334389,268334077)